### PR TITLE
`Initial_slot:initial_slot + number_slots` concern.

### PR DESCRIPTION
**Pull request by slowpoke1214**
This pull request should highlight our `Initial_slot:initial_slot + number_slots` changes for context to our question @carlosnatalino .

### CODE REVIEW

File: optical_rl_gym/envs/rmcsa_env.py [hunk 1]
```diff
@@ -320,14 +337,20 @@ def get_number_slots(self, path: Path) -> int:
...
+        Method that determines if the path is free for the core, path, and initial_slot.
+        """
         if initial_slot + number_slots > self.num_spectrum_resources:
             # logging.debug('error index' + env.parameters.rsa_algorithm)
             return False
         for i in range(len(path.node_list) - 1):
             if np.any(self.topology.graph['available_slots'][
-                      self.topology[path.node_list[i]][path.node_list[i + 1]]['index'],
-                      initial_slot:initial_slot + number_slots] == 0):
+                      initial_slot:initial_slot + number_slots,
```
Comment (slowpoke1214) [inline optical_rl_gym/envs/rmcsa_env.py:351]: @carlosnatalino Here's one instance.

File: optical_rl_gym/envs/rmcsa_env.py [hunk 2]
```diff
@@ -162,18 +174,18 @@ def reset(self, only_counters=True):
...
             raise ValueError("Path {} has not enough capacity on slots {}-{}".format(path.node_list, path, initial_slot,
                                                                                      initial_slot + number_slots))
 
         self.logger.debug('{} assigning path {} on initial slot {} for {} slots'.format(self.service.service_id, path.node_list, initial_slot, number_slots))
         for i in range(len(path.node_list) - 1):
-            self.topology.graph['available_slots'][self.topology[path.node_list[i]][path.node_list[i + 1]]['index'],
-                                                                        initial_slot:initial_slot + number_slots] = 0
-            self.spectrum_slots_allocation[self.topology[path.node_list[i]][path.node_list[i + 1]]['index'],
-                                                    initial_slot:initial_slot + number_slots] = self.service.service_id
+            self.topology.graph['available_slots'][initial_slot:initial_slot + number_slots, core,
```
Comment (slowpoke1214) [inline optical_rl_gym/envs/rmcsa_env.py:185]: @carlosnatalino Here's the other instance of our change.

File: optical_rl_gym/envs/rmcsa_env.py (hunk 1 above)
Comment (slowpoke1214) [reply optical_rl_gym/envs/rmcsa_env.py:351]: If this path of implementation is correct, then we will be able to continue a significant amount of changes. Before we wanted to proceed though, we weren't sure if this was a proper implementation due to our questions.

File: optical_rl_gym/envs/rmcsa_env.py [hunk 3]
```diff
@@ -320,14 +337,20 @@ def get_number_slots(self, path: Path) -> int:
...
         if initial_slot + number_slots > self.num_spectrum_resources:
             # logging.debug('error index' + env.parameters.rsa_algorithm)
             return False
         for i in range(len(path.node_list) - 1):
             if np.any(self.topology.graph['available_slots'][
-                      self.topology[path.node_list[i]][path.node_list[i + 1]]['index'],
-                      initial_slot:initial_slot + number_slots] == 0):
+                      initial_slot:initial_slot + number_slots,
+                      core,
+                      self.topology[path.node_list[i]][path.node_list[i + 1]]['index']] == 0):
```
Comment (slowpoke1214) [inline optical_rl_gym/envs/rmcsa_env.py:353]: If this path of implementation is correct, then we will be able to continue a significant amount of changes. Before we wanted to proceed though, we weren't sure if this was a proper implementation due to our questions.

Comment (slowpoke1214) [reply optical_rl_gym/envs/rmcsa_env.py:353]: Our main issue came from this block of code. Trying to understand what this code does has been difficult, we understand its checking if any of the items inside equal to 0. What we're wondering about it though is for example, does each item (initial_slot, core, self.topology) perform an index operation on `self.topology.graph['available_slots']` and it checks if those evaluate to 0. Or does it check just if `core` it self evaluates to 0 and so on.

**Review by carlosnatalino — APPROVED** [review]
The changes to the init files and the addition of the RMCSA file are correct and therefore, approved.

File: optical_rl_gym/envs/rmcsa_env.py (hunk 2 above)
Comment (carlosnatalino) [reply optical_rl_gym/envs/rmcsa_env.py:185]: Note that in line 157, you define the `available_slots` as (spatial_resources, edges, spectrum_resources).

Therefore, the access to this matrix should always follow the same order. This means that the core should be indexed before the slots, i.e., `core` should come before link and slots.

**Pull request closed without merging**